Failing to compile nativelib should stop everything

Currently, failing the `nativelib` will just `println` an error message, but won't stop the whole process, which may still report success:

```
> tests/run
[info] Updating {file:/Users/martin/Documents/Projects.nosync/Duhemm/scala-native/}tests...
[info] Resolving jline#jline;2.14.3 ...
[info] Done updating.
[info] Compiling 5 Scala sources to /Users/martin/Documents/Projects.nosync/Duhemm/scala-native/unit-tests/target/scala-2.11/classes...
[warn] [0] /main/scala/scala/scalanative/IssuesSuite.scala:264: a pure expression does nothing in statement position; you may be omitting necessary parentheses
[warn] [1] /main/scala/scala/scalanative/IssuesSuite.scala:265: a pure expression does nothing in statement position; you may be omitting necessary parentheses
[warn] [2] unknown:0: there was one deprecation warning; re-run with -deprecation for details
[warn] /main/scala/scala/scalanative/IssuesSuite.scala: 264 [0], 265 [1]
[warn] unknown: 0 [2]
[error] /Users/martin/Documents/Projects.nosync/Duhemm/scala-native/unit-tests/target/scala-2.11/native/lib/statvfs.c:44:1: error: unknown type name 'foobar'
[error] foobar
[error] ^
[error] /Users/martin/Documents/Projects.nosync/Duhemm/scala-native/unit-tests/target/scala-2.11/native/lib/statvfs.c:46:1: error: expected identifier or '('
[error] int scalanative_fstatvfs(int fd, struct scalanative_statvfs *buf) {
[error] ^
[error] 2 errors generated.
Failed to compile native library runtime code.
[info] Linking (1635 ms)
[info] Discovered 2341 classes and 14739 methods
[info] Optimizing (8468 ms)
[info] Generating intermediate code (918 ms)
[info] Produced 53 files
[info] Compiling to native code (2914 ms)
[info] Linking native code (421 ms)
//
// Hundreds of lines
//
* tests.SuiteSuite$
  [ok] expects true
  [ok] does not expect false
  [ok] expects not false
  [ok] does not expect true
  [ok] expects A and throws A
  [ok] expects C and throws C with function
  [ok] expects A and throws B
  [ok] expects A and doesn't throw
[success] Total time: 17 s, completed May 3, 2017 9:51:53 AM
```

The error in `nativeCompileLib` is very unlikely to be noticed in this scenario.

--- a/sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala
+++ b/sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala
@@ -1,8 +1,6 @@
 package scala.scalanative
 package sbtplugin
 
-import util._
-
 import sbtcrossproject.CrossPlugin.autoImport._
 import ScalaNativePlugin.autoImport._
 
@@ -12,13 +10,11 @@ import scalanative.io.VirtualDirectory
 import scalanative.util.{Scope => ResourceScope}
 
 import sbt._, Keys._, complete.DefaultParsers._
-import xsbti.{Maybe, Reporter, Position, Severity, Problem}
-import KeyRanks.DTask
 
 import scala.util.Try
 
 import System.{lineSeparator => nl}
-import java.io.File
+import java.io.ByteArrayInputStream
 
 object ScalaNativePluginInternal {
 

--- a/sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala
+++ b/sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala
@@ -250,7 +250,7 @@ object ScalaNativePluginInternal {
             logger.running(compilec)
             val result = Process(compilec, cwd) ! logger
             if (result != 0) {
-              println("Failed to compile native library runtime code.")
+              sys.error("Failed to compile native library runtime code.")
             }
         }
       }

--- a/sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala
+++ b/sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala
@@ -96,22 +96,27 @@ object ScalaNativePluginInternal {
     }
 
   lazy val projectSettings =
-    unscopedSettings ++
-      inConfig(Compile)(externalDependenciesTask(compile)) ++
-      inConfig(Test)(externalDependenciesTask(compile in Test)) ++
-      inConfig(Compile)(availableDependenciesTask(compile)) ++
-      inConfig(Test)(availableDependenciesTask(compile in Test)) ++
-      inConfig(Compile)(nativeMissingDependenciesTask) ++
-      inConfig(Test)(nativeMissingDependenciesTask)
+    dependencies ++
+      inConfig(Compile)(scalaNativeSettings) ++
+      inConfig(Test)(scalaNativeSettings)
 
-  lazy val unscopedSettings = Seq(
+  lazy val scalaNativeSettings =
+    scopedSettings ++
+      externalDependenciesTask(compile) ++
+      availableDependenciesTask(compile) ++
+      nativeMissingDependenciesTask
+
+  lazy val dependencies = Seq(
     libraryDependencies ++= Seq(
       "org.scala-native" %%% "nativelib" % nativeVersion,
       "org.scala-native" %%% "javalib"   % nativeVersion,
       "org.scala-native" %%% "scalalib"  % nativeVersion
     ),
     addCompilerPlugin(
-      "org.scala-native" % "nscplugin" % nativeVersion cross CrossVersion.full),
+      "org.scala-native" % "nscplugin" % nativeVersion cross CrossVersion.full)
+  )
+
+  lazy val scopedSettings = Seq(
     nativeWarnOldJVM := {
       val logger = nativeLogger.value
       Try(Class.forName("java.util.function.Function")).toOption match {
@@ -184,13 +189,13 @@ object ScalaNativePluginInternal {
     },
     nativeMode := "debug",
     artifactPath in nativeLink := {
-      (crossTarget in Compile).value / (moduleName.value + "-out")
+      crossTarget.value / (moduleName.value + "-out")
     },
     nativeLinkerReporter := tools.LinkerReporter.empty,
     nativeOptimizerReporter := tools.OptimizerReporter.empty,
     nativeOptimizerDriver := tools.OptimizerDriver(nativeConfig.value),
     nativeWorkdir := {
-      val workdir = (Keys.crossTarget in Compile).value / "native"
+      val workdir = crossTarget.value / "native"
       IO.delete(workdir)
       IO.createDirectory(workdir)
       workdir
@@ -253,10 +258,10 @@ object ScalaNativePluginInternal {
       lib
     },
     nativeConfig := {
-      val mainClass = (selectMainClass in Compile).value.getOrElse(
+      val mainClass = selectMainClass.value.getOrElse(
         throw new MessageOnlyException("No main class detected.")
       )
-      val classpath = (fullClasspath in Compile).value.map(_.data)
+      val classpath = fullClasspath.value.map(_.data)
       val entry     = nir.Global.Top(mainClass.toString + "$")
       val cwd       = nativeWorkdir.value
 
@@ -375,7 +380,7 @@ object ScalaNativePluginInternal {
       nativeWarnOldJVM.value
       // We explicitly mention all of the steps in the pipeline
       // although only the last one is strictly necessary.
-      (compile in Compile).value
+      compile.value
       nativeLinkNIR.value
       nativeOptimizeNIR.value
       nativeGenerateLL.value

--- a/scripted-tests/run/execution-context/build.sbt
+++ b/scripted-tests/run/execution-context/build.sbt
@@ -5,7 +5,7 @@ scalaVersion := "2.11.11"
 lazy val runAndCheck = taskKey[Unit]("...")
 
 runAndCheck := {
-  val bin = nativeLink.value
+  val bin = (nativeLink in Compile).value
   val out = Process(bin.getAbsolutePath).lines_!.toList
   assert(
     out == List(

--- a/scripted-tests/run/linker-reporter/build.sbt
+++ b/scripted-tests/run/linker-reporter/build.sbt
@@ -5,7 +5,8 @@ enablePlugins(ScalaNativePlugin)
 
 scalaVersion := "2.11.11"
 
-nativeLinkerReporter := LinkerReporter.toFile(target.value / "out.dot")
+nativeLinkerReporter in Compile := LinkerReporter.toFile(
+  target.value / "out.dot")
 
 lazy val check = taskKey[Unit]("Check that dot file was created.")
 

--- a/scripted-tests/run/optimizer-reporter/build.sbt
+++ b/scripted-tests/run/optimizer-reporter/build.sbt
@@ -5,7 +5,8 @@ enablePlugins(ScalaNativePlugin)
 
 scalaVersion := "2.11.11"
 
-nativeOptimizerReporter := OptimizerReporter.toDirectory(crossTarget.value)
+nativeOptimizerReporter in Compile := OptimizerReporter.toDirectory(
+  crossTarget.value)
 
 lazy val check = taskKey[Unit]("Check that dot file was created.")
 

